tests/test_client_session.py

```diff
@@ -379,6 +380,42 @@ def test_context_manager(connector, loop):
     assert e.strerror == err.strerror
 
 
+async def test_close_conn_on_error(create_session):
+    class UnexpectedException(Exception):
```

Comment:
Your try/except block catches BaseExceptions, but here you test just for Exception. It's easy to change the code and break the behavior, but not the tests.

Reply:
Done.